cdap-docs/admin-manual/source/_includes/installation/configuration.txt

```diff
@@ -162,28 +162,56 @@ Guide <https://hbase.apache.org/book.html#ulimit>`__.
----------------------
-Temp directories, depending on the distribution, are utilized by CDAP (the first two of
-this set of directories are properties specified in the ``cdap-site.xml`` file, as
-described in the :ref:`appendix-cdap-site.xml`):
-
-- ``app.temp.dir`` (default: ``/tmp``)
-- ``kafka.log.dir`` (default: ``/tmp/kafka-logs``)
-- ``/var/cdap/run``
-- ``/var/log/cdap``
-- ``/var/run/cdap``
-- ``/var/tmp/cdap`` 
-
-The CDAP user (the ``cdap`` system user) **must** be able to write to these directories, as they
-are used for deploying applications and for operating CDAP.
-
-As in all installations, the ``kafka.log.dir`` may need to be created locally. If you
-configure ``kafka.log.dir`` (or any of the other settable parameters) to a particular
-directory, you need to make sure that **the directory exists** and that it **is writable**
-by the CDAP user.
+Local Storage Configuration
+---------------------------
+Local storage directories |---| depending on the distribution |---| are utilized
+by CDAP for deploying applications and operating CDAP.
+
+The CDAP user (the ``cdap`` system user) **must** be able to write to **all** of these
+directories, as they are used for deploying applications and for operating CDAP.
+  
+- **List of local storage directories**
+
+  - Properties specified in the ``cdap-site.xml`` file, as described in the :ref:`appendix-cdap-site.xml`:
+
+    - ``app.temp.dir`` (default: ``/tmp``)
+    - ``kafka.server.log.dirs`` (default: ``/tmp/kafka-logs``)
+    - ``local.data.dir`` (default: ``data``; if this is instead an absolute path, needs to be writable)
+
+  - Additional directories:
+
+    - ``/var/cdap/run`` (used as a PID directory, created by the packages)
+    - ``/var/log/cdap`` (used as log directory, created by the packages)
+    - ``/var/run/cdap`` (default CDAP user's home directory, created by the packages)
+    - ``/var/tmp/cdap`` (default ``LOCAL_DIR`` |---| see below |---| defined and created in the CDAP init scripts)
+
+- Note that ``local.data.dir`` |---| which defines the directory for program jar storage
+  when deploying to YARN |---| is set in the ``cdap-site.xml`` and defaults to the
+  relative path ``data``. If the value of ``local.data.dir`` is *relative,* it is put
+  under ``LOCAL_DIR``, such as ``/var/tmp/cdap/data``. However, if instead it is an
+  *absolute* path, that alone is used as the value. This is desirable so you can easily
+  configure this directory to be elsewhere.
+
+- The CDAP Master service is governed by environment variables, which set the
+  directories it uses:
+
+  - ``TEMP_DIR`` (default: ``/tmp``): The directory serving as the ``java.io.tmpdir``
+    directory
+
+  - ``LOCAL_DIR`` (default: ``/var/tmp/cdap``): The directory serving as the user directory
+    for CDAP Master
+
+  These variables can be set in a file ``/etc/cdap/conf/cdap-env.sh`` and will be used as
+  the environment when launching CDAP and programs. See :ref:`|distribution|-configuration-central` 
```

Comment:
"included in the environment" (or "sourced into the environment") instead of "used as the environment"